src/server.js

```diff
@@ -2,25 +2,22 @@ var express = require('express');
 var app = express();
 var bodyParser = require( 'body-parser' );
 
-const { addOrUpdateScores, fetchScores } = require("./mongo/mongo");
-const fs = require( 'fs' );
+const { addOrUpdateScores, findWithId } = require("./mongo/mongo");
+const { roleNames, rolesDb } = require("./common/roles");
+const { lcns, isValidScore } = require("./utils/util");
+
 const path = require('path');
-const rolesDir = path.join(process.cwd(), "src/roles");
 const hostname = "http://localhost:8080";
+
+var db;
```

Comment:
What is the purpose of this `db` is it a db instance? I think this is a risky pattern given this could be multiple threaded there are risks that multiple users could interfere with each other. Perhaps make this a local variable in scope of the function?

Reply:
Given my other comments, I think it should just fetch the roles from database each time - no need to optimise.